core/services/validators.go

```diff
@@ -205,7 +205,10 @@ func validateTask(task models.TaskSpec, store *store.Store) error {
 			return errors.New("EthTxABIEncode Adapter is not implemented yet")
 		}
 	}
-	return err
+	if err != nil {
```

Comment:
Should this be closer to `adapters.For`?

Reply:
I'd prefer it that way personally, but I didn't want to mess with the existing control flow.